npm/webpack-dev-server/src/makeWebpackConfig.ts

```diff
@@ -94,6 +92,42 @@ export function makeWebpackConfig (
     devServerEvents,
   } = config.devServerConfig
 
+  let configFile: string | undefined = undefined
+
+  if (!userWebpackConfig && !frameworkWebpackConfig) {
+    debug('Not user or framework webpack config received. Trying to automatically source it')
+
+    const { default: findUp } = await importModule('find-up')
+
+    configFile = await findUp(configFiles, { cwd: projectRoot } as { cwd: string })
+
+    if (configFile) {
+      debug('found webpack config %s', configFile)
+      const sourcedConfig = await importModule(configFile)
+
+      debug('config contains %o', userWebpackConfig)
+      if (sourcedConfig && typeof sourcedConfig === 'object') {
+        userWebpackConfig = sourcedConfig.default ?? sourcedConfig
+      }
+    }
+
+    if (!userWebpackConfig) {
+      debug('could not find webpack.config!')
+      if (config.devServerConfig?.onConfigNotFound) {
+        config.devServerConfig.onConfigNotFound('webpack', projectRoot, configFiles)
+        // The config process will be killed from the parent, but we want to early exit so we don't get
+        // any additional errors related to not having a config
+        process.exit(0)
```

Comment:
this does not look/feel right... will `ipc.send(...)` even resolve considering its async if you are immediately killing the process?

Reply:
It it does otherwise the system tests wouldn't be passing. Could log an issue to follow up on this, because changing this will require a bit more investigation/work.